src/datasources/balances-api/entities/zerion-balance.entity.spec.ts

```diff
@@ -0,0 +1,557 @@
+import {
+  zerionAttributesBuilder,
+  zerionBalanceBuilder,
+  zerionBalancesBuilder,
+  zerionFlagsBuilder,
+  zerionFungibleInfoBuilder,
+  zerionImplementationBuilder,
+  zerionQuantityBuilder,
+} from '@/datasources/balances-api/entities/__tests__/zerion-balance.entity.builder';
+import {
+  ZerionAttributesSchema,
+  ZerionBalanceSchema,
+  ZerionBalancesSchema,
+  ZerionFlagsSchema,
+  ZerionFungibleInfoSchema,
+  ZerionImplementationSchema,
+  ZerionQuantitySchema,
+} from '@/datasources/balances-api/entities/zerion-balance.entity';
+import { faker } from '@faker-js/faker';
+import { ZodError } from 'zod';
+
+describe('Zerion Balance Entity schemas', () => {
+  describe('ZerionBalancesSchema', () => {
+    it('should validate a ZerionBalances object', () => {
+      const zerionBalances = zerionBalancesBuilder().build();
+
+      const result = ZerionBalancesSchema.safeParse(zerionBalances);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should not allow invalid', () => {
+      const zerionBalances = {
+        data: 'invalid',
+      };
+
+      const result = ZerionBalancesSchema.safeParse(zerionBalances);
+
+      expect(!result.success && result.error).toStrictEqual(
+        new ZodError([
+          {
+            code: 'invalid_type',
+            expected: 'array',
+            received: 'string',
+            path: ['data'],
+            message: 'Expected array, received string',
+          },
+        ]),
+      );
+    });
+  });
+  describe('ZerionBalanceSchema', () => {
+    it('should validate a ZerionBalance object', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+
+      const result = ZerionBalanceSchema.safeParse(zerionBalance);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should not allow an invalid type value', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+      // @ts-expect-error - type is expected to be a 'positions' literal
+      zerionBalance.type = 'invalid';
+
+      const result = ZerionBalanceSchema.safeParse(zerionBalance);
+
+      expect(!result.success && result.error).toStrictEqual(
+        new ZodError([
+          {
+            received: 'invalid',
+            code: 'invalid_literal',
+            expected: 'positions',
+            path: ['type'],
+            message: 'Invalid literal value, expected "positions"',
+          },
+        ]),
+      );
+    });
+
+    it('should not allow an invalid id value', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+      // @ts-expect-error - id is expected to be a string
+      zerionBalance.id = faker.number.int();
```

Comment:
I was initially going to suggest we move this into the builder but this is a much more future-proof method of overwriting the values that we should consider doing elsewhere in schema tests - if the type changes to match that expected, TypeScript will complain. (If we casted it inside the `with` of a builder, it wouldn't.)

Reply:
It makes perfect sense to me! 👍🏻

Reply:
It makes perfect sense to me! 👍🏻